### DISCUSSION TITLE
How to ignore group by

### DISCUSSION THREAD

**Discussion by djouallah** [malloydata/malloy#622] [Q&A] [answered]
in this example
```
source: names_table is table('duckdb:lineitem.parquet') {
  measure : Quantity is L_QUANTITY.sum()
  measure: Quantity_RETURNFLAG_A is Quantity { where: L_RETURNFLAG = 'N' }

    Query: by_returnflag is {
    group_by : L_RETURNFLAG
    aggregate : Quantity
    aggregate : Quantity_RETURNFLAG_A
 }
```
it return this results

```
L_​RETURNFLAG	Quantity	            Quantity_​RETURNFLAG_​A	
N	                          774,486,807	    774,486,807	
R	                          377,732,830	       ∅	
A	                          377,518,399          ∅	
```

How to get Quantity_RETURNFLAG_A to give the same number regardless of the grouping ?

**Comment by lloydtabb** [comment, marked as answer]
I'm not sure I understand the question but is this what you're asking about?


https://looker-open-source.github.io/malloy/documentation/patterns/percent_of_total.html

**Reply by djouallah** [reply]
Thanks, but it is too complex, ideally something like sum All Quantity at the measure level will be really useful.

**Comment by lloydtabb** [comment]
Soon.  https://github.com/looker-open-source/malloy/pull/618

**Comment by djouallah** [comment]
Thanks, the measure Quantity_​RETURNFLAG_​A seems to return wrong results when using nested Query ?

![image](https://user-images.githubusercontent.com/12554469/180128217-3ac4f2cc-bc8c-4367-9b70-0144b3d58294.png)

**Reply by lloydtabb** [reply]
Thanks for the report..  I'm not able to reproduce your issue exactly, but I've found two bugs related (our duckdb implementation is early, this works on BigQuery).

sum() should never return NULL, always 0, much like count()

Our duckdb node driver isn't handing nulls in numbers correctly, it seems and crashing form (and maybe giving incorrect results for you).

try
```
  measure: Quantity is IFNULL(L_QUANTITY.sum(), 0)
```

**Reply by lloydtabb** [reply]
https://github.com/looker-open-source/malloy/issues/623

https://github.com/looker-open-source/malloy/issues/624

**Comment by djouallah** [comment]
Thanks
  `measure: Quantity is IFNULL(L_QUANTITY.sum(), 0)`
gave me the correct results
